fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/api/UserDetailsApiResource.java

```diff
@@ -58,9 +59,8 @@
 @Component
 @Profile("oauth")
 @Scope("singleton")
-
-@Tag(name = "Fetch authenticated user details", description = "")
-@SuppressWarnings("deprecation") // TODO FINERACT-1012
+@Tag(name = "User Details", description = "Ability to fetch Authenticated User Details")
+@EnableResourceServer
```

Comment:
I'm not super familiar with Spring Security, but it strikes me as curious that the `UserDetailsApiResource`, which must work but with and without OAuth, needs a `pringframework.security.oauth2` specific annotation. Is this required? What does this do?